plugins/pluginmanagers/lazy.nix

```
@@ -1,222 +1,204 @@
+    my7h3le
+  ];
 
-  processPlugin =
-    plugin:
+  extraOptions =
     let
-      mkEntryFromDrv =
-        p:
-        if lib.isDerivation p then
+      # A plugin defined in the `nixvim.plugins.lazy.plugins` list can either
+      # be of `types.package` or `types.str`. Depending on the type of the
+      # given plugin this plugin will conditionally return an appropriate
+      # plugin spec.
+      coerceToLazyPluginSpec =
+        plugin:
+        if lib.isDerivation plugin then
           {
-            name = "${lib.getName p}";
-            path = p;
+            dir = lib.mkDefault "${plugin}";
+            name = lib.mkDefault "${lib.getName plugin}";
           }
+        else if lib.isString plugin then
+          { __unkeyed = lib.mkDefault plugin; }
         else
-          {
-            name = "${lib.getName p.pkg}";
-            path = p.pkg;
-          };
-      processDependencies =
-        if plugin ? dependencies && plugin.dependencies != null then
-          builtins.concatMap processPlugin plugin.dependencies
-        else
-          [ ];
-    in
-    [ (mkEntryFromDrv plugin) ] ++ processDependencies;
-
-  processedPlugins = builtins.concatLists (builtins.map processPlugin lazyPlugins);
-  lazyPath = pkgs.linkFarm "lazy-plugins" processedPlugins;
-in
-{
-  options = {
-    plugins.lazy = {
-      enable = mkEnableOption "lazy.nvim";
+          plugin;
 
-      gitPackage = lib.mkPackageOption pkgs "git" {
-        nullable = true;
-      };
+      lazyPluginCoercableType = with types; either str package;
 
-      plugins =
+      lazyPluginType =
         with types;
-        let
-          pluginType = either package (submodule {
-            options = {
-              dir = helpers.mkNullOrOption str "A directory pointing to a local plugin";
-
-              pkg = mkOption {
-                type = package;
-                description = "Vim plugin to install";
-              };
+        types.coercedTo lazyPluginCoercableType coerceToLazyPluginSpec (
+          submodule (
+            { config, ... }:
+
+            let
+              cfg = config;
+            in
+            {
+              freeformType = attrsOf anything;
+              options = {
+                __unkeyed = mkNullOrOption str ''
+                  The "unkeyed" attribute is the plugin's short plugin url.
+                  It ill be expanded using `config.git.url_format`. It can
+                  also be a url or dir.
+                '';
+
+                dir = mkNullOrOption str "A directory pointing to a local plugin";
+
+                url = mkNullOrOption str "A custom git url where the plugin is hosted";
+
+                pkg = mkNullOrOption package "Vim plugin to install";
+
+                name = mkNullOrOption str "Name of the plugin to install";
+
+                dev = defaultNullOpts.mkBool false ''
+                  When true, a local plugin directory will be used instead.
+                  See config.dev
```

> See config.dev

What does this mean? Should we explain it more here or link to docs?

---

Alright, looking at those docs makes it more clear what it's doing. 

```suggestion
                  This is the directory configured at `plugins.lazy.settings.dev.path`.
```